### DISCUSSION TITLE
Migrating from fzf.vim

### DISCUSSION THREAD

**Discussion by sluongng** [ibhagwan/fzf-lua#1386] [General]
Hey I am a long-time FZF.vim user trying to migrate to fzf-lua. This is my dotfiles for reference https://github.com/sluongng/dotfiles/blob/396f17d84cdb4b3fe82625b904e454e05e645991/.vimrc#L290-L334

Here are a few problems I ran into that I am not entirely sure how to solve after 30 minutes playing around and ended up reverting my dotfiles.

- `:Commands` is not available. In fzf.vim, `:Commands` let you fuzzy search through all the commands available. From the README of fzf-lua, it seems like the lower case `:commands` should be available? But I could not get it to work. This topic is a bit hard to search for because `commands` means different things in different context... So I would appreciate any pointer/help available.

- `:Rg` works differently. In fzf.vim, if you run `:Rg foo`, it will let you fuzzy search through the grep results. This means that the entries are already narrowed down to all places containing `foo`. My usually workflow would involve searching for `:Rg SomeClass`, then navigate to the result I want by fuzzy search for the file extension: `.java` or `.proto`. In fzf-lua, I tried the same command and it would open up the entire world and let me fuzzy search for `SomeClass`. This makes it harder for me to add the file extension to the search input, the search is slower as well.

Cheers.

**Comment by ibhagwan** [comment]
> :commands should be available? But I could not get it to work.

```vim
:FzfLua commands
```

> :Rg works differently. In fzf.vim, if you run :Rg foo

The fzf.vim compat mode is just a “kickstart”, you’re better off learning the fzf-lua way, everything can be found in the wiki/ issues, to search for a specific term:

Try these and play around with ctrl-g until you get it
```vim
:FzfLua grep search=foo
:FzfLua live_grep search=foo.*bar no_esc=true
```

**Comment by sluongng** [comment]
Thanks. I did not realize that there are subcommand to `:FzfLua`. This is a good building brick for me to customize my dotfiles 🤗

**Reply by ibhagwan** [reply]
100% of what's possible with fzf.vim is possible with fzf-lua, most esoteric (i.e. undocumented) stuff can be found in the docs/Wiki.

Read [OPTIONS.md](https://github.com/ibhagwan/fzf-lua/blob/main/OPTIONS.md) to understand how to supply options / configure fzf-lua.

**Reply by ibhagwan** [reply]
FYI, https://github.com/ibhagwan/fzf-lua/commit/c2a0044c09e42cf43e7aa77994cce36fe846b80c this commit also added the missing `:Commands`.
